moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java

```diff
@@ -1069,9 +1068,14 @@ public BufferedSource nextSource() throws IOException {
       throw new JsonDataException("Expected a value but was " + peek() + " at path " + getPath());
     }
 
+    // Advance the path and clear peeked if we haven't already.
+    if (peeked != PEEKED_NONE) {
```

Comment:
this check is necessary to avoid double-incrementing the position in cases like unquoted strings.